awx/main/tests/data/inventory/plugins/gce/files/gcp_compute.yml

```diff
@@ -0,0 +1,46 @@
+auth_kind: serviceaccount
+compose:
+  ansible_ssh_host: networkInterfaces | json_query("[0].accessConfigs[0].natIP")
+  gce_description: description if description else None
+  gce_id: id
+  gce_machine_type: machineType
+  gce_metadata: metadata.get("items", []) | items2dict(key_name="key", value_name="value")
+  gce_name: name
+  gce_network: networkInterfaces | json_query("[0].network.name")
+  gce_private_ip: networkInterfaces | json_query("[0].networkIP")
+  gce_public_ip: networkInterfaces | json_query("[0].accessConfigs[0].natIP")
+  gce_status: status
+  gce_subnetwork: networkInterfaces | json_query("[0].subnetwork.name")
+  gce_tags: tags | json_query("items")
+  gce_zone: zone
+hostnames:
+- name
+- public_ip
+- private_ip
+keyed_groups:
+- key: networkInterfaces | json_query("[0].subnetwork.name")
```

Comment:
Alternatively use `gce_subnetwork` since it's already defined above and the templar evaluation doesn't have to happen twice.

Reply:
That's not possible, because of:

https://github.com/ansible/ansible/issues/51883

It's relatively trivial to duplicate the syntax twice here. Even if it could be fixed in the constructed plugin, that would constrain the order in which the inventory options are processed. Since it's the inventory plugin code that calls these methods, that becomes a very difficult proposition. It would need to be enforced for _each_ plugin, and that's just not worth it IMO.